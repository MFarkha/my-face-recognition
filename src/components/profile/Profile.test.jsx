import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Profile from './Profile';
const mockUser = {
    firstname: 'John',
    age: '42',
    pet: 'dog',
    joined: '1/24/2023',
    entries: 23
}

test('renders Profile container correctly for a given user', () => {
    const { container } = render(<Profile user={ mockUser } />);
    expect(screen.getByRole('heading', { name: mockUser.firstname }))
        .toHaveTextContent(mockUser.firstname);
    expect(screen.getByRole('heading', { name: /images/i }))
        .toHaveTextContent(`Images submitted: ${mockUser.entries}`);
    expect(screen.getByLabelText(/name/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/name/i)).toHaveAttribute('placeholder', mockUser.firstname);
    expect(screen.getByLabelText(/age/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/age/i)).toHaveAttribute('placeholder', mockUser.age);
    expect(screen.getByLabelText(/pet/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/pet/i)).toHaveAttribute('placeholder', mockUser.pet);
    expect(screen.getByRole('button', { name: /save/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /cancel/i })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});

test('cancels updates to a profile for a given user', async () => {
    const mockToggle = jest.fn();
    render(<Profile user={ mockUser } toggleModal={ mockToggle }/>);
    const user = userEvent.setup();
    await user.click(screen.getByRole('button', { name: /cancel/i }))
    expect(mockToggle).toHaveBeenCalled();
})

// test('updates a profile for a given user', async () => {
//     // const mockFetch = jest.fn()
//     // .mockReturnValue(Promise.resolve({
//     //     json: () => Promise.resolve({
//     //         count: 87,
//     //         results: [1,2,3,4]
//     //     })
//     // }));
//     // expect.assertions(4);
//     const mockToggle = jest.fn();

//     render(<Profile user={ mockUser } toggleModal={ mockToggle }/>);
//     const user = userEvent.setup();
//     await user.click(screen.getByRole('button', { name: /save/i }))
//     expect(mockToggle).toHaveBeenCalled();
// })