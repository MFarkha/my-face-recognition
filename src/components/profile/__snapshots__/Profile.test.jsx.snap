// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile container correctly for a given user 1`] = `
<div>
  <div
    class="profile-modal"
  >
    <article
      class="br3 ba b--black-10 mv4 w-100 w-50-m w-25-l mw6 shadow-5 center bg-white"
    >
      <main
        class="pa4 black-80 w-80"
      >
        <img
          alt="avatar"
          class="h3 w3 dib"
          src="profile.png"
        />
        <h1>
          John
        </h1>
        <h4>
          Images submitted: 23
        </h4>
        <p>
          Member since: 1/24/2023
        </p>
        <label
          class="mt-2 fw-6"
          for="user-name"
        >
          First Name
        </label>
        <input
          class="pa2 ba w-100"
          id="user-name"
          name="user-name"
          placeholder="John"
          type="text"
        />
        <label
          class="mt-2 fw-6"
          for="user-age"
        >
          Age
        </label>
        <input
          class="pa2 ba w-100"
          id="user-age"
          name="user-age"
          placeholder="42"
          type="text"
        />
        <label
          class="mt-2 fw-6"
          for="user-pet"
        >
          Pet
        </label>
        <input
          class="pa2 ba w-100"
          id="user-pet"
          name="user-pet"
          placeholder="dog"
          type="text"
        />
        <div
          class="mt4"
          style="display: flex; justify-content: space-evenly;"
        >
          <button
            class="b pa2 grow pointer hover-white w-40 bg-light-blue b--black-20"
            name="save"
          >
            Save
          </button>
          <button
            class="b pa2 grow pointer hover-white w-40 bg-light-red b--black-20"
            name="cancel"
          >
            Cancel
          </button>
        </div>
      </main>
      <div
        class="modal-close"
      >
        Ã—
      </div>
    </article>
  </div>
</div>
`;
